File: src/main/java/com/beat/domain/performance/api/HomeController.java
package com.beat.domain.performance.api;

import com.beat.domain.performance.application.HomeService;
import com.beat.domain.performance.application.dto.home.HomeFindRequest;
import com.beat.domain.performance.application.dto.home.HomeFindAllResponse;
import com.beat.domain.performance.domain.Genre;
import com.beat.domain.performance.exception.PerformanceSuccessCode;
import com.beat.global.common.dto.SuccessResponse;

import lombok.RequiredArgsConstructor;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/main")
@RequiredArgsConstructor
public class HomeController implements HomeApi {

	private final HomeService homeService;

	@Override
	@GetMapping
	public ResponseEntity<SuccessResponse<HomeFindAllResponse>> getHomePerformanceList(
		@RequestParam(required = false) Genre genre) {

		HomeFindRequest homeFindRequest = new HomeFindRequest(genre);

		HomeFindAllResponse homeFindAllResponse = homeService.findHomePerformanceList(homeFindRequest);
		return ResponseEntity.ok(
			SuccessResponse.of(PerformanceSuccessCode.HOME_PERFORMANCE_RETRIEVE_SUCCESS, homeFindAllResponse));
	}
}

File: apps/api/src/main/kotlin/com/beat/apps/api/home/api/HomeController.kt
package com.beat.apps.api.home.api

import com.beat.apps.api.home.api.response.HomeFindAllResponse
import com.beat.apps.api.home.api.response.HomeSuccessCode
import com.beat.apps.api.home.api.type.HomeGenreType
import com.beat.apps.api.home.facade.HomeFacade
import com.beat.apps.api.response.SuccessResponse
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/main")
class HomeController(
    private val homeFacade: HomeFacade,
) : HomeApi {

    @GetMapping
    override fun getHomePerformanceList(
        @RequestParam(required = false) genre: HomeGenreType?,
    ): ResponseEntity<SuccessResponse<HomeFindAllResponse>> {
        val homeFindAllResponse = homeFacade.findHomePerformanceList(genre)
        return ResponseEntity.ok(
            SuccessResponse.of(HomeSuccessCode.HOME_PERFORMANCE_RETRIEVE_SUCCESS, homeFindAllResponse),
        )
    }
}
